### ISSUE TITLE
downloading from updated links is now a pain

### ISSUE THREAD

**Issue by trucman** [tonikelope/megabasterd#327]
Heya,

megabasterd used to detecte files already downloaded and skip them it was useful.
now it detects files already present, but recreate a file with random characters, simply put a duplicate.

it's particularly painfull when you download from a link with content updated regularly (with tons of files).

could this become a choice, I can see how having files re-downloaded could be a good thing, but it seems both solutions should be options.

**Comment by tonikelope** [comment]
Yes, I will add it as an option and leave the old default behavior. Sorry for the inconvenience.

**Comment by trucman** [comment]
thanks, your program is awesome by the way ;)

**Comment by tonikelope** [comment]
I have modified it so that in case of detecting the same file (name and size) it stops. If the same file exists but with another size it generates the new random one. For the moment it will stay like this until next updates that I will evaluate if I will add an option in settings. I hope it will help you.

**Comment by trucman** [comment]
cool beans, it makes things so much easier. thank you.

**Comment by trucman** [comment]
heya,

after some time trying the new version, here's few suggestions, still related to pre-existing files.

now the programs checks for file names and size to see if it's the same. when it finds an existing files, it says so, and let a line saying that files already exist.

problem : it makes browsing through download list difficult if there's many files that were already present, and you have to delete these manually. when it's a thousand files, it takes some time to clean the download list.
when I say it makes browsing difficult, it's in case of error while downloading.
also I'm not sure but it seems that if the dowload list is massive (thousand files), it affects the downloads (stopping) if there's too many files still in the job list (even if it's saying it's existing files)

suggestion : a button to delete files that were detected as duplicates. or at least a way to order the job list by error message and maybe being able to select them by batch and delete by different means other than clicking "close" on each file.

best regards

**Comment by ericziethen** [comment]
@trucman , a workaround for this is to download anything else, and once that is finished the "clear finished" button will appear on top which will clear the existing files as well.

**Comment by trucman** [comment]
yeah I know, it didn't work back then, it would just sit there until you manually deleted them. it's a nice improvement.
